src/interpolations.jl

```diff
@@ -451,7 +451,7 @@ function reference_coordinates(ip::DiscontinuousLagrange{RefTetrahedron,0,T}) wh
 end
 
 function shape_value(ip::DiscontinuousLagrange{shape, 0}, ::Vec{dim, T}, i::Int) where {dim, shape <: AbstractRefShape{dim}, T}
-    i > 1 && throw(ArgumentError("no shape function $i for interpolation $ip"))
+    i > 1 && throw(BoundsError(ip, i))
```

Comment:
Just noting it here, since I looked here how you did for using it somewhere else. 
I'm not sure if the error message is informative enough.
```
ERROR: BoundsError: attempt to access Lagrange{RefQuadrilateral, 1, Nothing} at index [1]
```

Reply:
If there is a better way to do it in a type stable way I am happy to implement it. I chose `BoundsError` over `ArgumentsError` because the latter is incompatible with evaluating individual shape functions on the GPU due to the apparent type instability.

Reply:
BoundsError feels wrong here though. Is ArgumentError with a const string (no interpolation) also problematic?

Reply:
Why? The evaluation here acts like a lazy array access for the last index.

I remember that is also was an issue with constant strings, but I cannot quickly reproduce the issue in 1.9.3. Maybe this one has been fixed.